app/services/fact_importer.rb

```diff
@@ -111,7 +115,11 @@ def normalize(facts)
     facts.keep_if { |k, v| v.present? }
   end
 
+  def db_facts_as_relation
+    host.fact_values.eager_load(:fact_name).where("fact_names.type = '#{fact_name_class}'")
+  end
+
   def db_facts
-    @db_facts ||= host.fact_values.eager_load(:fact_name).where("fact_names.type = '#{fact_name_class}'").index_by(&:name)
+    @db_facts ||= db_facts_as_relation.index_by(&:name)
```

Comment:
The only use AFAICT for this function is above in `add_new_facts` - where we only pick out the names anyways. 
So I think it could be rewritten as `db_facts_as_relation.pluck('fact_names.name')` and this method and instance variable can be discarded. 
only other consideration is if other plugins depend on this function, not sure.